apps/sms/js/thread_list_ui.js

```diff
@@ -659,13 +659,6 @@ var ThreadListUI = {
       // Register all threads to the Threads object.
       Threads.set(thread.id, thread);
 
-      // If one of the requested threads is also the currently displayed thread,
-      // update the header immediately
-      // TODO: Revise necessity of this code in bug 1050823
-      if (Navigation.isCurrentPanel('thread', { id: thread.id })) {
-        ThreadUI.updateHeaderData();
-      }
```

Comment:
I think https://github.com/mozilla-b2g/gaia/blob/392ef2fafa48166378fb18e33da1ab4e04235e66/apps/sms/js/thread_ui.js#L537 didn't really made this code useless, but "Threads.registerMessage" in the right places did :)

Since ```updateHeaderData``` relies on the fact that active thread is defined. There are 2 cases when we can "easily" break it - accidentally remove ```Threads.registerMessage``` call from the code path for run-as-notification or run-as-activity (since we delay thread list rendering for these cases) - that is why I want to have integration tests for these cases